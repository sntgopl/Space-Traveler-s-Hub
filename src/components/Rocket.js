import PropTypes from 'prop-types';
import { Button, Badge } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { reserveRocket, cancelRocket } from '../redux/rockets/rockets';
import '../stylesheets/Rocket.css';

const Rocket = ({
  id, name, img, description, reserved,
}) => {
  const dispatch = useDispatch();
  return (
    <div className="rocket">
      <img src={img} alt="rocket" />
      <div>
        <h2>{name}</h2>
        <p>
          {' '}
          {reserved && (
          <Badge bg="info">Reserved</Badge>
          )}
          {' '}
          {description}

        </p>
        {!reserved && (
          <Button onClick={() => dispatch(reserveRocket(id))} variant="primary">Reserve Rocket</Button>
        )}
        {reserved && (
          <Button onClick={() => dispatch(cancelRocket(id))} variant="outline-danger">Cancel Reservation</Button>
        )}
      </div>
    </div>
  );
};

export default Rocket;

Rocket.propTypes = {
  id: PropTypes.node.isRequired,
  name: PropTypes.node.isRequired,
  img: PropTypes.node.isRequired,
  description: PropTypes.node.isRequired,
  reserved: PropTypes.bool.isRequired,
};
