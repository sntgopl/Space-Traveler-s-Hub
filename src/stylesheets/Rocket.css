.rocket {
  display: flex;
  gap: 20px;
  padding: 30px 0;
}

.rocket img {
  width: 300px;
}
