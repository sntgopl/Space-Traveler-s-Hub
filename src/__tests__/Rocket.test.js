import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Rocket from '../components/Rocket';
import store from '../redux/configureStore';

describe('test snapshot', () => {
  it('should render Rocket Container', () => {
    const renderRocketContainer = renderer.create(
      <Provider store={store}>
        <Rocket
          key={1}
          id={1}
          name="Rocket test"
          type="Rocker"
          img="url test"
          description="It is a rocket"
          reserved={false}
        />
      </Provider>,
    ).toJSON();
    expect(renderRocketContainer).toMatchSnapshot();
  });
});

describe('test with testing library', () => {
  it('should change to cancel reservation', () => {
    render(
      <Provider store={store}>
        <Rocket
          key={1}
          id={1}
          name="Rocket test"
          type="Rocker"
          img="url test"
          description="It is a rocket"
          reserved
        />
      </Provider>,
    );
    expect(screen.getByText('Cancel Reservation')).toBeInTheDocument();
  });

  it('should change show reserved', () => {
    render(
      <Provider store={store}>
        <Rocket
          key={1}
          id={1}
          name="Rocket test"
          type="Rocker"
          img="url test"
          description="It is a rocket"
          reserved
        />
      </Provider>,
    );
    expect(screen.getByText('Reserved')).toBeInTheDocument();
  });

  it('should change to reserve rocket', () => {
    render(
      <Provider store={store}>
        <Rocket
          key={1}
          id={1}
          name="Rocket test"
          type="Rocker"
          img="url test"
          description="It is a rocket"
          reserved={false}
        />
      </Provider>,
    );
    expect(screen.getByText('Reserve Rocket')).toBeInTheDocument();
  });
});
