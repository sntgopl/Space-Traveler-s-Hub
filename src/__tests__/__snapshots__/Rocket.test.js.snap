// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot should render Rocket Container 1`] = `
<div
  className="rocket"
>
  <img
    alt="rocket"
    src="url test"
  />
  <div>
    <h2>
      Rocket test
    </h2>
    <p>
       
       
      It is a rocket
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
